core/src/main/scala/io/delta/tables/DeltaTable.scala

```diff
@@ -472,6 +473,24 @@ class DeltaTable private[tables](
   def upgradeTableProtocol(readerVersion: Int, writerVersion: Int): Unit = {
     deltaLog.upgradeProtocol(Protocol(readerVersion, writerVersion))
   }
+
+  /**
+   * Restores delta table to specific version.
+   * @param version version to be restored
+   * @since 1.2.0
+   */
+  def restore(version: Long): Unit = {
+    executeRestore(deltaLog, version = Some(version))
+  }
+
+  /**
+   * Restores delta table to the latest commit that happened at or before `timestamp`.
+   * @param timestamp timestamp to be restored
+   * @since 1.2.0
+   */
+  def restore(timestamp: Timestamp): Unit = {
```

Comment:
I think eventually we would like to add a SQL API which would take the timestamp as a string. so to mimic that and avoid having two different code paths, do you mind taking the timestamp as a string? 

the string can be of the format yyyy-MM-dd or yyyy-MM-dd HH:mm:ss